bitcoin/src/taproot.rs

```diff
@@ -112,7 +118,12 @@ impl TapNodeHash {
             eng.input(b.as_ref());
             eng.input(a.as_ref());
         };
-        TapNodeHash::from_engine(eng)
+        (TapNodeHash::from_engine(eng), a < b)
+    }
+
+    /// Assumes the given 32 byte array as hidden [`TapNodeHash`].
+    pub fn assume_hidden(hash: [u8; 32]) -> TapNodeHash {
```

Comment:
Given a method for constructing from array exists, what is the purpose of this? Just better readability? Perhaps write a note about it in the doc comment?